#838 - Apply new FSPIOP error handling to central ledger code base

gibaros opened the pull request:
- replace old throw Error for new FSPIOPError
- merge mojaloop master
- fix unit tests

rmothilal commented:
Please update package.json with the latest dependencies and version and run npm install and commit

rmothilal commented:
Please update package.json with the latest dependencies and version and run npm install and commit

gibaros commented:
PR currently under review, QA please note the following 2 API changes:

1. at transfers/handler.js line 333 an error code change, previously code thrown was 3300, now is throwing 3303

2. at position/facade.js line 222, error description changed as 4001 error description was ‘Payer FSP has insufficient liquidity to perform the transfer’ which is different from new API error description ‘Payer FSP insufficient liquidity’

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -404,7 +404,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve({ enumeration: 'ABORTED' }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -426,7 +426,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve({ enumeration: 'RECEIVED' }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -445,7 +445,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve({ enumeration: 'unknown' }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -465,7 +465,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve({ enumeration: 'RESERVED' }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -484,7 +484,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: true
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -504,7 +504,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: true
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -523,7 +523,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -541,7 +541,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -560,7 +560,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -578,7 +578,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -596,7 +596,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -615,7 +615,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -626,7 +626,7 @@ Test('Transfer handler', transferHandlerTest => {
       Util.transformAccountToTopicName.returns(topicName)
       Util.proceed.returns(true)
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -645,7 +645,7 @@ Test('Transfer handler', transferHandlerTest => {
       Util.transformAccountToTopicName.returns(topicName)
       Util.proceed.returns(true)
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -672,7 +672,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -691,7 +691,7 @@ Test('Transfer handler', transferHandlerTest => {
         existsNotMatching: false
       }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -703,7 +703,7 @@ Test('Transfer handler', transferHandlerTest => {
         Util.transformAccountToTopicName.returns(topicName)
         Util.proceed.returns(true)
-        Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+        // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -927,7 +927,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.getById.returns(P.resolve(null))
       TransferService.validateDuplicateHash.returns(P.resolve({}))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -941,7 +941,7 @@ Test('Transfer handler', transferHandlerTest => {
       Util.transformGeneralTopicName.returns(topicName)
       TransferService.getById.returns(P.resolve(null))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -956,7 +956,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.getById.returns(P.resolve(null))
       TransferService.validateDuplicateHash.returns(P.resolve({}))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -970,7 +970,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.getById.returns(P.resolve({ payeeFsp: 'dfsp2' }))
       TransferService.validateDuplicateHash.returns(P.resolve({}))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -986,7 +986,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.getById.returns(P.resolve({ payeeFsp: 'dfsp2' }))
       TransferService.validateDuplicateHash.returns(P.resolve({}))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -1001,7 +1001,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.getById.returns(P.resolve({ condition: 'condition', payeeFsp: 'dfsp2' }))
       TransferService.validateDuplicateHash.returns(P.resolve({}))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -1018,7 +1018,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.getById.returns(P.resolve({ condition: 'condition', payeeFsp: 'dfsp2' }))
       TransferService.validateDuplicateHash.returns(P.resolve({}))
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please remove commented out code.

gibaros replied:
Removed

mdebarros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -1035,7 +1035,7 @@ Test('Transfer handler', transferHandlerTest => {
       TransferService.validateDuplicateHash.returns(P.resolve({}))
       Validator.validateFulfilCondition.returns(true)
-      Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
+      // Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)
```
Please apply the same comment for ALL similiarly commented out code in this file --> `// Util.createPrepareErrorStatus.returns(fulfilMessages[0].value.content.payload)`

gibaros replied:
All of these occurrences have been removed

mdebarros commented on test/unit/lib/errors.test.js:
```diff
@@ -26,8 +26,7 @@
 
```
Please delete this test as it is no longer applicable.

gibaros replied:
Deleted

mdebarros commented:
> PR currently under review, QA please note the following 2 API changes:
> 
> 1. at transfers/handler.js line 333 an error code change, previously code thrown was 3300, now is throwing 3303
> 2. at position/facade.js line 222, error description changed as 4001 error description was ‘Payer FSP has insufficient liquidity to perform the transfer’ which is different from new API error description ‘Payer FSP insufficient liquidity’

@sri-miriyala @NicoDuvenage FYI

gibaros commented:
> Looks good to me

Thank you

gibaros commented:
> nd version an

Thank you, update to 7.2.2 has been performed and pushed, will continue with other changes in review.

gibaros commented on .gitignore:
```diff
@@ -54,5 +54,5 @@ jspm_packages
 **/*.iml
 
-#package-lock.json
+package-lock.json
```
package-lock.json entry has been removed from .gitignore

gibaros commented on src/handlers/lib/dao.js:
```diff
@@ -48,9 +49,10 @@ const retrieveAllParticipants = async () => {
-    throw e
+  } catch (err) {
+    Logger.error(err)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration and just throw the fspiopError

gibaros commented on src/handlers/lib/kafka/consumer.js:
```diff
@@ -98,7 +99,8 @@ const getConsumer = (topicName) => {
     return listOfConsumers[topicName].consumer
   } else {
-    throw Error(`No consumer found for topic ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No consumer found for topic ${topicName}`)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/kafka/consumer.js:
```diff
@@ -116,7 +118,8 @@ const isConsumerAutoCommitEnabled = (topicName) => {
     return listOfConsumers[topicName].autoCommitEnabled
   } else {
-    throw Error(`No consumer found for topic ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No consumer found for topic ${topicName}`)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on package-lock.json:
```diff
@@ -2063,6 +2063,15 @@
         "safer-buffer": "^2.1.0"
```
npm install was ran after updating the central-services-error-handling and package.json and its corresponding lock file are in sync

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -88,8 +90,8 @@ const disconnect = async (topicName = null) => {
     } catch (err) {
-      Logger.error(err)
-      throw err
+      const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -105,10 +107,12 @@ const disconnect = async (topicName = null) => {
     }
     if (isError) {
-      throw Error(`The following Producers could not be disconnected: ${JSON.stringify(errorTopicList)}`)
+      const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`The following Producers could not be disconnected: ${JSON.stringify(errorTopicList)}`)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -126,7 +130,8 @@ const getProducer = (topicName) => {
     return listOfProducers[topicName]
   } else {
-    throw Error(`No producer found for topic ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No producer found for topic ${topicName}`)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -70,7 +71,8 @@ const produceMessage = async (messageProtocol, topicConf, config) => {
   } catch (err) {
     Logger.error(err)
     Logger.info(`Producer error has occurred for ${topicConf.topicName}`)
```
// TODO - gibaros - review if the message inside the logger info can be added to error message...if worth it

mdebarros replied:
Let's discuss this once you are online today.

gibaros replied:
Fixed per review

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -105,10 +107,12 @@ const disconnect = async (topicName = null) => {
     }
   } else {
-    throw Error(`Unable to disconnect Producer: ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`Unable to disconnect Producer: ${topicName}`)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -153,9 +154,10 @@ const participantTopicTemplate = (participantName, functionality, action) => {
-    throw e
+  } catch (err) {
+    Logger.error(err)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -195,8 +198,8 @@ const transformGeneralTopicName = (functionality, action) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -300,16 +302,16 @@ const updateMessageProtocolMetadata = (messageProtocol, metadataType, metadataAc
-    }
-  }
-}
+// const createPrepareErrorStatus = (errorCode, errorDescription, extensionList) => {
```
Removed

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -541,7 +542,7 @@ module.exports = {
   getKafkaConfig,
   updateMessageProtocolMetadata,
-  createPrepareErrorStatus,
+  // createPrepareErrorStatus,
```
Removed

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -503,20 +505,19 @@ const breadcrumb = (location, message) => {
-    metadataState = createState(ENUMS.STATE.FAILURE.status, code, desc)
+
+    message.value.content.payload = fspiopError
+    metadataState = createState(ENUMS.STATE.FAILURE.status, fspiopError.errorInformation.errorCode, fspiopError.errorInformation.errorDescription)
```
Need to review if createState() should be removed and if so, how to populate the metadataState send on the state parameter to the produceGeneralMessage()

gibaros replied:
No change needed per review with Miguel

gibaros commented on src/handlers/positions/handler.js:
```diff
@@ -86,7 +83,8 @@ const positions = async (error, messages) => {
   if (error) {
     // Logger.error(error)
-    throw error
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(error)
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/positions/handler.js:
```diff
@@ -199,31 +194,30 @@ const positions = async (error, messages) => {
-        errorInformation.extensionList = payload.extensionList
-        return await Util.proceed(params, { consumerCommit, histTimerEnd, errorInformation, producer })
+        const fspiopError = ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.EXPIRED_ERROR).toApiErrorObject()
+        fspiopError.extensionList = payload.extensionList
```
extensionList passed as a param at createFSPIOPError call, removed unnecessary additional step

gibaros commented on src/handlers/positions/handler.js:
```diff
@@ -247,7 +241,8 @@ const registerPositionHandler = async () => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/positions/handler.js:
```diff
@@ -263,8 +258,8 @@ const registerAllHandlers = async () => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/register.js:
```diff
@@ -75,7 +76,8 @@ const registerAllHandlers = async () => {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
+    throw fspiopError
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/timeouts/handler.js:
```diff
@@ -111,7 +110,8 @@ const timeout = async () => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/timeouts/handler.js:
```diff
@@ -165,7 +165,8 @@ const registerTimeoutHandler = async () => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/timeouts/handler.js:
```diff
@@ -184,8 +185,8 @@ const registerAllHandlers = async () => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -127,9 +127,9 @@ const prepare = async (error, messages) => {
       if (!transferState) {
         Logger.error(Util.breadcrumb(location, `callbackErrorNotFound1--${actionLetter}1`))
-        const errorInformation = Errors.getErrorInformation(errorType.internal, 'transfer/state not found')
+        const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError().toApiErrorObject()
```
'transfer/state not found' passed as message to createInternalServerFSPIOPError()

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -172,20 +172,21 @@ const prepare = async (error, messages) => {
     Logger.error(`${Util.breadcrumb(location)}::${err.message}--P0`)
     histTimerEnd({ success: false, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId })
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Review if message at: Logger.error(`${Util.breadcrumb(location)}::${err.message}--P0`) should be added to FSPIOPError

gibaros replied:
Change requested at review has been performed

gibaros replied:
Fixed to add logger error message to FSPIOPError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -198,7 +199,8 @@ const fulfil = async (error, messages) => {
   if (error) {
     // Logger.error(error)
-    throw error
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(error)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -242,18 +244,18 @@ const fulfil = async (error, messages) => {
       if (!transferState) {
         Logger.error(Util.breadcrumb(location, `callbackErrorNotFound2--${actionLetter}1`))
-        const errorInformation = Errors.getErrorInformation(errorType.internal, 'transfer/state not found')
+        const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError().toApiErrorObject()
```
'transfer/state not found' added to createInternalServerFSPIOPError() as message

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -302,34 +304,35 @@ const fulfil = async (error, messages) => {
       } else if (existingTransfer.expirationDate <= new Date()) {
         Logger.info(Util.breadcrumb(location, `callbackErrorTransferExpired--${actionLetter}11`))
-        const errorInformation = Errors.getErrorInformation(errorType.transferExpired)
+        // TODO: - previously code thrown was 3300, now is throwing 3303
```
Node added at PR description and you added @ to specific QA team members after I added note

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -348,21 +351,22 @@ const fulfil = async (error, messages) => {
             const producer = { functionality: TransferEventType.POSITION, action: TransferEventAction.ABORT }
-            const errorInformation = Errors.getErrorInformation(TER.errorCode, { replace: TER.errorDescription })
-            return await Util.proceed(params, { consumerCommit, histTimerEnd, errorInformation, producer, toDestination })
+            const fspiopError = ErrorHandler.Factory.createFSPIOPError(ErrorHandler.FindFSPIOPErrorCode(TER.errorCode), TER.errorDescription).toApiErrorObject()
```
Changed to:

            const fspiopError = ErrorHandler.Factory.createFSPIOPErrorFromErrorCode(TER.errorCode, TER.errorDescription).toApiErrorObject()

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -348,21 +351,22 @@ const fulfil = async (error, messages) => {
     Logger.error(`${Util.breadcrumb(location)}::${err.message}--F0`)
     histTimerEnd({ success: false, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId })
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Review if message at: Logger.error(${Util.breadcrumb(location)}::${err.message}--P0) should be added to FSPIOPError

gibaros replied:
Change requested at review has been performed

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -384,7 +388,8 @@ const getTransfer = async (error, messages) => {
   if (error) {
     // Logger.error(error)
-    throw error
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(error)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -436,7 +441,8 @@ const getTransfer = async (error, messages) => {
     Logger.error(`${Util.breadcrumb(location)}::${err.message}--G0`)
     histTimerEnd({ success: false, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId })
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Review if message at: Logger.error(${Util.breadcrumb(location)}::${err.message}--P0) should be added to FSPIOPError

gibaros replied:
Change requested at review has been performed

gibaros replied:
Adding logger error message to FSPIOPError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -460,7 +466,8 @@ const registerPrepareHandler = async () => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -484,7 +491,8 @@ const registerFulfilHandler = async () => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed error logger so it is just thrown and logged where it is handled and also remove fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -507,8 +515,8 @@ const registerGetTransferHandler = async () => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/transfers/handler.js:
```diff
@@ -527,8 +535,8 @@ const registerAllHandlers = async () => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/shared/setup.js:
```diff
@@ -46,10 +45,11 @@ const RequestLogger = require('../lib/requestLogger')
 const UrlParser = require('../lib/urlParser')
 const Logger = require('@mojaloop/central-services-shared').Logger
-const Boom = require('boom')
+// const Participant = require('../domain/participant')
```
removed

gibaros commented on test/integration/helpers/transferDuplicateCheck.js:
```diff
@@ -87,6 +88,6 @@ exports.prepareData = async () => {
 //       await TransferPreparationModule.deletePreparedData(transferId, payerName, payeeName)
```
removed

gibaros commented on test/unit/handlers/lib/kafka/producer.test.js:
```diff
@@ -177,7 +177,7 @@ Test('Producer', producerTest => {
         test.fail('Error not thrown!')
       } catch (e) {
-        test.ok(e.toString() === 'Error: No producer found for topic undefined')
+        test.equal(e.message, 'No producer found for topic undefined')
```
Producer.getProducer() throws the following:

throw ErrorHandler.Factory.createInternalServerFSPIOPError(`No producer found for topic ${topicName}`)

so instead of adding:

test.assert(error instanceof Errors.ValidationError)

added:

test.assert(e instanceof FSPIOPError)

gibaros commented on test/unit/handlers/lib/kafka/producer.test.js:
```diff
@@ -258,7 +258,7 @@ Test('Producer', producerTest => {
       } catch (e) {
         test.ok(e instanceof Error)
-        test.ok(e.toString() === `Error: The following Producers could not be disconnected: [{"topic":"${topicNameFailure}","error":"No producer found for topic ${topicNameFailure}"}]`)
+        test.equal(e.message, `The following Producers could not be disconnected: [{"topic":"${topicNameFailure}","error":"No producer found for topic ${topicNameFailure}"}]`)
```
Producer.getProducer() throws the following:

throw ErrorHandler.Factory.createInternalServerFSPIOPError('')

so instead of adding:

test.assert(error instanceof Errors.ValidationError)

added:

test.assert(e instanceof FSPIOPError)

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -172,9 +174,10 @@ const participantTopicTemplate = (participantName, functionality, action) => {
-    throw e
+  } catch (err) {
+    Logger.error(err)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -215,8 +218,8 @@ const transformAccountToTopicName = (participantName, functionality, action) =>
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros commented on test/unit/handlers/lib/utility.test.js:
```diff
@@ -345,20 +346,20 @@ Test('Utility Test', utilityTest => {
-
-    createPrepareErrorStatusTest.end()
-  })
+  // utilityTest.test('createPrepareErrorStatus should', createPrepareErrorStatusTest => {
```
Removed

gibaros commented on test/unit/handlers/positions/handler.test.js:
```diff
@@ -669,7 +669,7 @@ Test('Position handler', transferHandlerTest => {
         config.rdkafkaConf['enable.auto.commit'] = false
         Util.transformGeneralTopicName.returns(topicName)
-        Util.createPrepareErrorStatus.returns(topicName)
+        // Util.createPrepareErrorStatus.returns(topicName)
```
Removed

gibaros commented on test/unit/handlers/positions/handler.test.js:
```diff
@@ -241,7 +241,7 @@ Test('Position handler', transferHandlerTest => {
       await Kafka.Consumer.createHandler(topicName, config, command)
       Util.transformGeneralTopicName.returns(topicName)
-      Util.createPrepareErrorStatus.returns(topicName)
+      // Util.createPrepareErrorStatus.returns(topicName)
```
Removed

gibaros commented on test/unit/handlers/positions/handler.test.js:
```diff
@@ -268,7 +268,7 @@ Test('Position handler', transferHandlerTest => {
         await Kafka.Consumer.createHandler(topicName, config, command)
         Util.transformGeneralTopicName.returns(topicName)
-        Util.createPrepareErrorStatus.returns(topicName)
+        // Util.createPrepareErrorStatus.returns(topicName)
```
Removed

gibaros commented on test/unit/handlers/positions/handler.test.js:
```diff
@@ -644,7 +644,7 @@ Test('Position handler', transferHandlerTest => {
         await Kafka.Consumer.createHandler(topicName, config, command)
         Util.transformGeneralTopicName.returns(topicName)
-        Util.createPrepareErrorStatus.returns(topicName)
+        // Util.createPrepareErrorStatus.returns(topicName)
```
Removed

gibaros commented on test/unit/handlers/positions/handler.test.js:
```diff
@@ -693,7 +693,7 @@ Test('Position handler', transferHandlerTest => {
         Kafka.Consumer.getConsumer.throws(new Error())
         Util.transformGeneralTopicName.returns(topicName)
-        Util.createPrepareErrorStatus.returns(topicName)
+        // Util.createPrepareErrorStatus.returns(topicName)
```
Removed

gibaros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -326,7 +326,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve(null))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Removed

gibaros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -346,7 +346,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve(null))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Removed

gibaros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -365,7 +365,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve(null))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Removed

gibaros commented on test/unit/handlers/transfers/handler.test.js:
```diff
@@ -383,7 +383,7 @@ Test('Transfer handler', transferHandlerTest => {
       }))
       TransferService.getTransferStateChange.withArgs(transfer.transferId).returns(P.resolve({ enumeration: 'COMMITTED' }))
-      Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
+      // Util.createPrepareErrorStatus.returns(messageProtocol.content.payload)
```
Removed

gibaros commented on package.json:
```diff
@@ -70,7 +70,7 @@
     "@mojaloop/central-object-store": "7.1.0",
```
Made sure central-object-store was set to 7.1.0 at package.json and ran npm install

gibaros commented on src/api/participants/handler.js:
```diff
@@ -99,7 +98,8 @@ const create = async function (request, h) {
     return h.response(entityItem(participant, ledgerAccountIds)).code(201)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -121,30 +121,31 @@ const createHubAccount = async function (request, h) {
     return h.response(entityItem(participant, ledgerAccountIds)).code(201)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -176,7 +177,8 @@ const update = async function (request) {
     return entityItem(updatedEntity, ledgerAccountIds)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -186,7 +188,8 @@ const addEndpoint = async function (request, h) {
     return h.response().code(201)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -217,7 +220,8 @@ const getEndpoint = async function (request) {
     }
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -227,7 +231,8 @@ const addLimitAndInitialPosition = async function (request, h) {
     return h.response().code(201)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -250,7 +255,8 @@ const getLimits = async function (request) {
     return limits
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -274,7 +280,8 @@ const getLimitsForAllParticipants = async function (request) {
     return limits
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -294,7 +301,8 @@ const adjustLimits = async function (request, h) {
     return h.response(updatedLimit).code(200)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -303,7 +311,8 @@ const getPositions = async function (request) {
     return await ParticipantService.getPositions(request.params.name, request.query)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -312,7 +321,8 @@ const getAccounts = async function (request) {
     return await ParticipantService.getAccounts(request.params.name, request.query)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -330,7 +340,8 @@ const updateAccount = async function (request, h) {
     return h.response().code(200)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/api/participants/handler.js:
```diff
@@ -341,7 +352,8 @@ const recordFunds = async function (request, h) {
     return h.response().code(202)
   } catch (err) {
-    throw Boom.badRequest(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/cryptoConditions/index.js:
```diff
@@ -1,13 +1,14 @@
-  } catch (error) {
-    throw new Errors.ValidationError(error.message)
+  } catch (err) {
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -55,14 +54,15 @@ const ParticipantAccountMismatchText = 'Participant/account mismatch'
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -74,8 +74,8 @@ const getAll = async () => {
   } catch (err) {
-    Logger.error(err)
-    throw new Error(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -114,8 +114,8 @@ const update = async (name, payload) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -124,8 +124,8 @@ const createParticipantCurrency = async (participantId, currencyId, ledgerAccoun
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -134,17 +134,17 @@ const createHubAccount = async (participantId, currencyId, ledgerAccountTypeId)
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -134,17 +134,17 @@ const createHubAccount = async (participantId, currencyId, ledgerAccountTypeId)
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -156,8 +156,8 @@ const destroyByName = async (name) => {
   } catch (err) {
-    Logger.error(err)
-    throw new Error(err.message)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -185,8 +185,8 @@ const addEndpoint = async (name, payload) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -212,8 +212,8 @@ const getEndpoint = async (name, type) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -238,8 +238,8 @@ const getAllEndpoints = async (name) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -261,8 +261,8 @@ const destroyParticipantEndpointByName = async (name) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -308,8 +308,8 @@ const addLimitAndInitialPosition = async (participantName, limitAndInitialPositi
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -329,8 +329,8 @@ const getPositionByParticipantCurrencyId = async (participantCurrencyId) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -350,8 +350,8 @@ const getPositionChangeByParticipantPositionId = async (participantPositionId) =
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -373,8 +373,8 @@ const destroyParticipantPositionByNameAndCurrency = async (name, currencyId) =>
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -397,8 +397,8 @@ const destroyParticipantLimitByNameAndCurrency = async (name, currencyId) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -432,8 +432,8 @@ const getLimits = async (name, { currency = null, type = null }) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -455,8 +455,8 @@ const getLimitsForAllParticipants = async ({ currency = null, type = null }) =>
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -493,8 +493,8 @@ const adjustLimits = async (name, payload) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -597,8 +597,8 @@ const getPositions = async (name, query) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -623,8 +623,8 @@ const getAccounts = async (name, query) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -635,34 +635,34 @@ const updateAccount = async (payload, params, enums) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -635,34 +635,34 @@ const updateAccount = async (payload, params, enums) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -635,34 +635,34 @@ const updateAccount = async (payload, params, enums) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/participant/index.js:
```diff
@@ -739,8 +739,8 @@ const recordFundsInOut = async (payload, params, enums) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -38,37 +38,33 @@ const TransferDuplicateCheckModel = require('../../models/transfer/transferDupli
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -84,29 +80,29 @@ const fulfil = async (transferFulfilmentId, transferId, payload) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -84,29 +80,29 @@ const fulfil = async (transferFulfilmentId, transferId, payload) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -84,29 +80,29 @@ const fulfil = async (transferFulfilmentId, transferId, payload) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -151,8 +147,8 @@ const validateDuplicateHash = async (transferId, payload, transferFulfilmentId =
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -177,8 +173,8 @@ const logTransferError = async (transferId, errorCode, errorDescription) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/admin/handler.js:
```diff
@@ -62,7 +63,8 @@ const createRecordFundsInOut = async (payload, transactionTimestamp, enums) => {
       } catch (err) {
         await trx.rollback
-        throw err
+        const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/admin/handler.js:
```diff
@@ -74,7 +76,8 @@ const createRecordFundsInOut = async (payload, transactionTimestamp, enums) => {
       } catch (err) {
         await trx.rollback
-        throw err
+        const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/admin/handler.js:
```diff
@@ -177,7 +180,8 @@ const transfer = async (error, messages) => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/admin/handler.js:
```diff
@@ -201,7 +205,8 @@ const registerTransferHandler = async () => {
   } catch (err) {
     Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/admin/handler.js:
```diff
@@ -218,8 +223,8 @@ const registerAllHandlers = async () => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/lib/dao.js:
```diff
@@ -48,9 +49,10 @@ const retrieveAllParticipants = async () => {
-    throw e
+  } catch (err) {
+    Logger.error(err)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/handlers/lib/kafka/consumer.js:
```diff
@@ -98,7 +99,8 @@ const getConsumer = (topicName) => {
     return listOfConsumers[topicName].consumer
   } else {
-    throw Error(`No consumer found for topic ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No consumer found for topic ${topicName}`)
```
Fixed per review.

gibaros commented on src/handlers/lib/kafka/consumer.js:
```diff
@@ -116,7 +118,8 @@ const isConsumerAutoCommitEnabled = (topicName) => {
     return listOfConsumers[topicName].autoCommitEnabled
   } else {
-    throw Error(`No consumer found for topic ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No consumer found for topic ${topicName}`)
```
Fixed per review.

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -70,7 +71,8 @@ const produceMessage = async (messageProtocol, topicConf, config) => {
     Logger.error(err)
     Logger.info(`Producer error has occurred for ${topicConf.topicName}`)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed per review.

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -38,37 +38,33 @@ const TransferDuplicateCheckModel = require('../../models/transfer/transferDupli
   const transfer = await TransferFacade.getById(id)
   if (!transfer) {
-    throw new Errors.TransferNotFoundError()
+    throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, 'This transfer does not exist')
```
Ready to update ADD_PARTY_INFO_ERROR to TRANSFER_ID_NOT_FOUND upon @mdebarros  signal

gibaros replied:
getFulfillment() has been removed and this code with it, no change needed

gibaros commented on src/domain/transfer/index.js:
```diff
@@ -38,37 +38,33 @@ const TransferDuplicateCheckModel = require('../../models/transfer/transferDupli
   const transferFulfilment = await TransferFulfilmentModel.getByTransferId(id)
   if (!transferFulfilment) {
-    throw new Errors.TransferNotFoundError()
+    throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, 'This transfer does not exist')
```
Ready to update ADD_PARTY_INFO_ERROR to TRANSFER_ID_NOT_FOUND upon @mdebarros  signal

gibaros replied:
getFulfillment() has been removed and this code with it, no change needed

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -70,7 +71,8 @@ const produceMessage = async (messageProtocol, topicConf, config) => {
     Logger.error(err)
     Logger.info(`Producer error has occurred for ${topicConf.topicName}`)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Removed fspiopError declaration to just throw the fspiopError

gibaros replied:
Should Logger.info be logged as error and/or be included at FSPIOPError?

gibaros replied:
Updated to:

    throw ErrorHandler.Factory.createInternalServerFSPIOPError = (`Producer error has occurred for ${topicConf.topicName}`, err)

gibaros commented:
@mdebarros @rmothilal All addressed reviews were set to resolved, corresponding code changes were performed, unit and integration tests passed. Please review the non-resolved reviews and their comments, thank you

mdebarros commented on test/integration/helpers/transferDuplicateCheck.js:
```diff
@@ -78,16 +78,3 @@ exports.prepareData = async () => {
   }
 }
-
-// exports.deletePreparedData = async (transferId, payerName, payeeName) => {
```
I am not sure this should be removed. I suspect it may be useful. I am also not sure why its commented out to be honest.

gibaros replied:
Re-added it, but still commented out.

gibaros commented on src/shared/setup.js:
```diff
@@ -103,9 +103,10 @@ const createServer = (port, modules) => {
+          options: ErrorHandler.validateRoutes(),
           failAction: async (request, h, err) => {
-            throw Boom.boomify(err)
+            const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed

gibaros commented on src/shared/setup.js:
```diff
@@ -257,7 +258,8 @@ const initialize = async function ({ service, port, modules = [], runMigrations
     default:
       Logger.error(`No valid service type ${service} found!`)
-      throw new Error(`No valid service type ${service} found!`)
+      const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No valid service type ${service} found!`)
```
Fixed

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -88,8 +90,8 @@ const disconnect = async (topicName = null) => {
     } catch (err) {
-      Logger.error(err)
-      throw err
+      const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -105,10 +107,12 @@ const disconnect = async (topicName = null) => {
     }
     if (isError) {
-      throw Error(`The following Producers could not be disconnected: ${JSON.stringify(errorTopicList)}`)
+      const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`The following Producers could not be disconnected: ${JSON.stringify(errorTopicList)}`)
```
Fixed

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -105,10 +107,12 @@ const disconnect = async (topicName = null) => {
     }
   } else {
-    throw Error(`Unable to disconnect Producer: ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`Unable to disconnect Producer: ${topicName}`)
```
Fixed

gibaros commented on src/handlers/lib/kafka/producer.js:
```diff
@@ -126,7 +130,8 @@ const getProducer = (topicName) => {
     return listOfProducers[topicName]
   } else {
-    throw Error(`No producer found for topic ${topicName}`)
+    const fspiopError = ErrorHandler.Factory.createInternalServerFSPIOPError(`No producer found for topic ${topicName}`)
```
Fixed

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -153,9 +154,10 @@ const participantTopicTemplate = (participantName, functionality, action) => {
-    throw e
+  } catch (err) {
+    Logger.error(err)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -172,9 +174,10 @@ const participantTopicTemplate = (participantName, functionality, action) => {
-    throw e
+  } catch (err) {
+    Logger.error(err)
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -195,8 +198,8 @@ const transformGeneralTopicName = (functionality, action) => {
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed

gibaros commented on src/handlers/lib/utility.js:
```diff
@@ -215,8 +218,8 @@ const transformAccountToTopicName = (participantName, functionality, action) =>
   } catch (err) {
-    Logger.error(err)
-    throw err
+    const fspiopError = ErrorHandler.Factory.reformatFSPIOPError(err)
```
Fixed

gibaros commented on src/models/ledgerAccountType/ledgerAccountType.js:
```diff
@@ -30,10 +30,6 @@
-    throw new Error(err.message)
-  }
+  const ledgerAccountType = await Db.ledgerAccountType.findOne({ name })
+  return ledgerAccountType
```
Updated per review

gibaros commented on src/models/position/facade.js:
```diff
@@ -119,10 +119,9 @@ const prepareChangeParticipantPositionTransaction = async (transferList) => {
-              errorInformation: Errors.createErrorInformation(4001, rawMessage.value.content.payload.extensionList)
-            }
+            // TODO: - review error description as 4001 error description is 'Payer FSP has insufficient liquidity to perform the transfer' which is different from new API 'Payer FSP insufficient liquidity'
+            transferState.reason = 'Payer FSP has insufficient liquidity to perform the transfer'
```
Fixed per review

gibaros commented on src/models/transfer/facade.js:
```diff
@@ -411,7 +394,7 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
         participants.push(participant)
       } else {
-        throw new Error('Invalid FSP name or currency')
+        throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.INTERNAL_SERVER_ERROR, 'Invalid FSP name or currency')
```
No change needed per review with Miguel and Rajiv...if validation of FSP needs to happen validation needs to happen at domain.
